### PR TITLE
 docs for Open IoT SDK #42 

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
ready for review

### CODE REVIEW

File: examples/lock-app/openiotsdk/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+The example showcases an implementation of the Door Lock cluster.
+
+## Build and run
+
+For information on how to build and run this example and further information
+about the platform it is run on see
+[Open IoT SDK examples](../../../docs/examples/openiotsdk_examples.md).
+
+The example name to use in the scripts is `lock-app`.
```
Comment (ATmobica) [inline examples/lock-app/openiotsdk/README.md:11]: I think we should provide details on how to test this application (commissioning + ZCL commands), and the expected output from it.

File: examples/shell/openiotsdk/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+For more details see
+[Common shell commands](../README.md#chip-shell-command-details).
+
+## Build and run
+
+For information on how to build and run this example and further information
+about the platform it is run on see
+[Open IoT SDK examples](../../../docs/examples/openiotsdk_examples.md).
+
+The example name to use in the scripts is `shell`.
```
Comment (ATmobica) [inline examples/shell/openiotsdk/README.md:17]: I think we should provide details on how to test this application - the expected output and example of a shell command.

File: docs/examples/openiotsdk_examples.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+You can call the script directly yourself.
+
+```
+${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -s <example name>
+```
+
+Use `--help` to get more information about the script options.
+
+## Running
```
Comment (ATmobica) [inline docs/examples/openiotsdk_examples.md:60]: We can mention using telnet connection in running step.

File: docs/guides/openiotsdk_commissioning.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Commissioning Open IoT SDK devices
+
+CHIP devices reset into into a ready for commissioning state. This allows a
+controller to connect to them and set configuration options before the device
+becomes available on the CHIP network.
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply docs/guides/openiotsdk_commissioning.md:5]: When the device boots it waits for the controller to receive config. After that it becomes part of the mesh and can receive normal commands. No?

File: src/platform/openiotsdk/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Open IoT SDK platform port
+
+This platform is based on
+[Open IoT SDK](https://gitlab.arm.com/iot/open-iot-sdk). Open IoT-SDK is a
+reference implementation of [Open-CMSIS-CDI](https://www.open-cmsis-cdi.org/)
+which defines a common device interface for microcontroller-based devices used
+in the Internet of Things. It is delivered as a framework of software components
+with a set of feature-rich example applications.
+
+## Targets
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/platform/openiotsdk/README.md]: Is this enough?

File: examples/lock-app/openiotsdk/README.md (hunk 1 above)
Comment (paul-szczepanek-arm) [reply examples/lock-app/openiotsdk/README.md:15]: the common examples links to commissioning readme, is that enough?

File: examples/shell/openiotsdk/README.md (hunk 2 above)
Comment (paul-szczepanek-arm) [reply examples/shell/openiotsdk/README.md:17]: the common examples links to commissioning readme, is that enough?

File: examples/lock-app/openiotsdk/README.md (hunk 1 above)
Comment (paul-szczepanek-arm) [reply examples/lock-app/openiotsdk/README.md:15]: otherwise do you have the zcl for lock-app?

File: examples/shell/openiotsdk/README.md (hunk 2 above)
Comment (ATmobica) [reply examples/shell/openiotsdk/README.md:17]: We don't use the commissioning here...
I think about something like this:

```
## Testing

Communication with the application goes through active telnet session.
When the application runs these lines should be visible:

   [INF] [SH] Open IoT SDK shell example application start
   [INF] [SH] Open IoT SDK shell example application run
   >

The shell application launched correctly. Pass commands to the terminal and wait for the
response.
The application supports common Matter shell commands. They are used to control
the basic functionalities of the device.

For more details visit:
[Common shell commands](../README.md#chip-shell-command-details)

Example:

    > echo Hello
    Hello
    Done
```

File: examples/lock-app/openiotsdk/README.md (hunk 1 above)
Comment (ATmobica) [reply examples/lock-app/openiotsdk/README.md:15]: I think about something like this:

```
## Testing

### Telent terminal
Communication with the application goes through an active telnet session.
When the application runs these lines should be visible:

   [INF] [-] Open IoT SDK lock-app example application start
   ...
   [INF] [-] Open IoT SDK lock-app example application run

The lock-app application launched correctly and you can follow traces in the
terminal.

### Matter tools
Read the [Open IoT SDK commissioning](../../../docs/guides/openiotsdk_commissioning.md) to
see how to use the Matter controller to commission and control the application.

The application fully supports the DoorLock cluster. Use its commands to trigger actions on the device.
Example:
In Matter controller:
   `zcl  DoorLock LockDoor 1234 1 pinCode=str:12345`
Device response:
   `[INF] [ZC] Lock App: specified PIN code was found in the database, setting door lock state to "Locked" [endpointId=1]`

File: src/platform/openiotsdk/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ATmobica) [reply src/platform/openiotsdk/README.md]: Yes

**Review by ATmobica — APPROVED** [review]

**Pull request merged**